### PR TITLE
Sync Sequential.compile() with Model.compile()

### PR DESCRIPTION
**Pull request by ozabluda**
This is a backward-compatible API change (API additions only).

### CODE REVIEW

File: keras/models.py [hunk 1]
```diff
@@ -741,28 +741,59 @@ def save_weights(self, filepath, overwrite=True):
 
     def compile(self, optimizer, loss,
                 metrics=None,
+                loss_weights=None,
```
Comment (fchollet) [inline keras/models.py]: Loss weights do not make sense for `Sequential` since the model has a single output.

File: keras/models.py [hunk 2]
```diff
@@ -741,28 +741,59 @@ def save_weights(self, filepath, overwrite=True):
...
                 sample_weight_mode=None,
                 weighted_metrics=None,
+                target_tensors=None,
```
Comment (fchollet) [inline keras/models.py]: We should consider naming it `target_tensor` since it will always be a single tensor.

File: keras/models.py (hunk 1 above)
Comment (ozabluda) [reply keras/models.py]: Maybe we should warn/assert that len(loss_weight)==1. Maybe somebody wants to multiply loss by a constant or something. It' mainly to make a sequential Model and Sequential interchangeable as much as practicable.

File: keras/models.py (hunk 2 above)
Comment (ozabluda) [reply keras/models.py]: Same as above. Maybe we should warn/assert that len(target_tensors)==1, to make a sequential Model and Sequential interchangeable

File: keras/models.py (hunk 1 above)
Comment (ozabluda) [reply keras/models.py]: When passing multiple loss_weights to this function, a very clear exception is raised:
```
  File "./examples/cifar10_cnn.py", line 68, in <module>
    loss_weights=[1,2])
  File "build/bdist.linux-x86_64/egg/keras/models.py", line 817, in compile
  File "build/bdist.linux-x86_64/egg/keras/engine/training.py", line 697, in compile
ValueError: When passing a list as loss_weights, it should have one entry per model
outputs. The model has 1 outputs, but you passed loss_weights=[1, 2]
```

File: keras/models.py (hunk 2 above)
Comment (ozabluda) [reply keras/models.py]: When passing multiple target_tensors to this function, a very clear exception is raised:
```
Traceback (most recent call last):
  File "./examples/cifar10_cnn.py", line 68, in <module>
    target_tensors=[1,2])
  File "build/bdist.linux-x86_64/egg/keras/models.py", line 817, in compile
  File "build/bdist.linux-x86_64/egg/keras/engine/training.py", line 715, in compile
ValueError: When passing a list as `target_tensors`, it should have one entry per model
outputs. The model has 1 outputs, but you passed target_tensors=[1, 2]
```

Comment (ozabluda) [reply keras/models.py]: It' mainly to make a sequential Model and Sequential interchangeable as much as practicable. For a sequential Model compile() we (correctly) don't prohibit loss_weights, and (correctly) have target_tensors. When passing multiple loss_weights or target_tensors, a very clear exception is raised:
```The model has 1 outputs, but you passed loss_weights=[1, 2]``` and 
```The model has 1 outputs, but you passed target_tensors=[1, 2]```

**Comment by ozabluda** [comment]
It's mainly to make a sequential Model and Sequential interchangeable as much as practicable. For a sequential Model compile() we (correctly) don't prohibit loss_weights, and (correctly) have target_tensors. When passing multiple loss_weights or target_tensors, a very clear exception is raised:
```The model has 1 outputs, but you passed loss_weights=[1, 2]``` and 
```The model has 1 outputs, but you passed target_tensors=[1, 2]```

File: keras/models.py (hunk 1 above)
Comment (fchollet) [reply keras/models.py]: I would definitely exclude `loss_weights` from the `Sequential` API. There is no use case for it, and it is likely to confuse some people.

File: keras/models.py (hunk 2 above)
Comment (fchollet) [reply keras/models.py]: Ok for this change.

File: keras/models.py (hunk 1 above)
Comment (ozabluda) [reply keras/models.py]: Excluded. While at it, I (accidentally) removed previously present Sequential.loss_weights attribute as well, but it seems like the right thing to do.

**Comment by ozabluda** [comment]
Clone/Open to restart tests

**Comment by ozabluda** [comment]
@fchollet, what should we do with Sequential.loss_weights attribute? If it really is confusing, maybe it should be made private (API change)? Or the fact that it was there all this time, is the additional argument sufficient to put loss_weights into the Sequential API?

**Comment by fchollet** [comment]
Since it isn't documented, it doesn't hurt anyone. No reason to remove it: the removal wouldn't be an improvement to anyone's experience of Keras.

**Pull request merged**